### DISCUSSION TITLE
How can I fetch a JPEG image using axios in React Native?

### DISCUSSION THREAD

**Discussion by hamzakat** [axios/axios#5388] [Q&A]
Hi all. I am trying to fetch an image from the server through axios but I couldn't. I tried to add this to the request configs `responseType: "blob"` but it doesn't work too. I can get the image normally through Postman, so I don't think the issue is from the server.

What is the proper way to achieve that on React Native?

**Comment by jasonsaayman** [comment]
react native has a couple issues with axios. please try searching the issues to see if there is anything config related you may need to implement.

as for an example here is some pseudo code:

```js
import axios from 'axios';

async function fetchAndDisplayImage() {
  try {
    const response = await axios.get('https://httpbin.org/image/png', {
      responseType: 'blob' // Important: Treats data as a binary object
    });

    // Create a temporary local URL for the blob
    const imageObjectURL = URL.createObjectURL(response.data);

    // Set the source of an image element in your HTML
    const imgElement = document.getElementById('my-image');
    imgElement.src = imageObjectURL;

  } catch (error) {
    console.error('Error fetching image:', error);
  }
}

fetchAndDisplayImage();
```
